src/components/Application.vue

```diff
@@ -0,0 +1,251 @@
+<template>
+  <div id="app">
+    <GmapMap
+      :center="{lat: initialPosition.lat, lng:initialPosition.lng}"
+      :zoom="10"
+      map-type-id="terrain"
+      style="width: 100%; height: 90%"
+    >
+      <!-- check if icon link in makers payload then display -->
+      <GmapMarker
+        :key="index"
+        v-for="(m, index) in markers"
+        :position="m.position"
+        :clickable="true"
+        :draggable="false"
+        @click="center=m.position"
+        :icon="m.icon"
+        :title="m.userName"
+      />
+    </GmapMap>
+
+    <div class="notification">
+      <p>Online Users: {{markers.length}}</p>
+        <ul>
+          <li v-for="(user, i) in onlineUsers" :key="i">
+            <pre v-text="user.data.userName"></pre>
+          </li>
+        </ul>
+    </div>
+  </div>
+</template>
+
+<script>
+import axios from "axios";
+import * as Ably from "ably";
+import Loading from "vue-loading-overlay";
+import "vue-loading-overlay/dist/vue-loading.css";
+
+var ably = new Ably.Realtime({
+  key: "<ably key>",
+  clientId: `${Math.random() * 1000000}`
+});
+
+export default {
+  name: "Application",
+  mounted() {
+    const name = prompt('To get started, input your name in the field below and locate your friends around based on your location, please turn on your location setting \n What is your name?')
+    this.usersName = name
+    const channel = prompt('Enter name of channel you are interested in')
+    this.channelName = channel
+  },
+  async created() {
+    await this.fetchData();
+    var channel = ably.channels.get(this.channelName);
+    console.log(channel)
+    await channel.attach(err => {
+      if (err) {
+        return console.error("Error attaching to the channel");
+      }
+      console.log("We are now attached to the channel");
+      channel.presence.enter(this.userlocation, function(err) {
+        if (err) {
+          return console.error("Error entering presence");
+        }
+        console.log("We are now successfully present");
+      });
+    });
+
+    let membersData;
+    channel.presence.subscribe('update',function(presenceMsg) {
+      console.log(
+        "Received a " + presenceMsg.action + " from " + presenceMsg.clientId
+      );
+      channel.presence.get(function(err, members) {
+        membersData = members
+        console.log(
+          "There are now " + members.length + " clients present on this channel"
+        );
+      });
+    });
+
+
+
+    this.polling = setInterval(() => {
+      this.onlineUsers = membersData
+      this.markers = membersData.map((mem) => {
+        if (JSON.stringify(this.userlocation) == JSON.stringify(mem.data)) {
+          return {...mem.data, icon: 'http://maps.google.com/mapfiles/ms/icons/blue-dot.png'}
```

Comment:
Syntax Error: Unexpected token (88:18) here. This error doesn't let me test the app generally, please fix it.

Reply:
i don't understand the error, you are getting

Reply:
let me look at it again

Reply:
fixed, you can have a look now